tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx

```diff
@@ -94,6 +120,53 @@ public:
       bool IsEmpty() const { return fFieldZero->begin() == fFieldZero->end(); }
    };
 
+   /// A model is usually immutable after giving up on it, e.g. to construct a `RNTupleWriter`. However, for the rare
+   /// cases that require changing the model after the fact, `RIncrementalUpdater` provides limited support for
+   /// incremental updates, e.g. addition of new fields.
+   ///
+   /// See `RNTupleWriter::GetIncrementalModelUpdater()` for an example.
+   class RIncrementalUpdater {
+      friend class RNTupleWriter;
+
+   private:
+      RNTupleWriter &fWriter;
+      Detail::RNTupleModelChangeset fOpenChangeset;
+
+      RIncrementalUpdater(RNTupleWriter &writer);
+
+   public:
+      ~RIncrementalUpdater() { CommitUpdate(); }
+      /// Begin a new set of alterations to the underlying model. As a side effect, all `REntry` instances related to
+      /// the model are invalidated.
+      void BeginUpdate();
+      /// Commit changes since the last call to `BeginUpdate()`. All the invalidated `REntry`s remain invalid.
+      /// `CreateEntry()` or `CreateBareEntry()` can be used to create an `REntry` that matching the new model.
+      /// Upon completion, `BeginUpdate()` can be called again to begin a new set of changes.
+      void CommitUpdate();
+
+      void AddField(std::unique_ptr<Detail::RFieldBase> field);
+
+      // TODO(jalopezg): `MakeField<T>` should also be probably added for convenience
+      template <typename T>
```

Comment:
I think that it may be actually useful to have these two `AddField()` overloads as-is, as they provide a means to also capture a value -- avoiding an additional call to `CaptureValueUnsafe()`.